helix-view/src/view.rs

```diff
@@ -98,11 +94,20 @@ impl View {
             last_accessed_doc: None,
             last_modified_docs: [None, None],
             object_selections: Vec::new(),
+            gutter_types: gutters,
         }
     }
 
-    pub fn gutters(&self) -> &[(Gutter, usize)] {
-        GUTTERS
+    pub fn gutters(&self) -> Vec<(Gutter, usize)> {
+        let mut gutters: Vec<(Gutter, usize)> = vec![];
+        use crate::editor::GutterType;
+        for gutter in &self.gutter_types {
+            match gutter {
+                GutterType::Diagnostics => gutters.push((gutter::diagnostics_or_breakpoints, 1)),
+                GutterType::LineNumbers => gutters.push((gutter::line_numbers, 5)),
+            }
+        }
+        gutters
     }
```

Comment:
Rather than here, can we do this during config deserialization so that config already stores `Vec<Gutter, usize>`? Current code will allocate a new vec and do a bunch of looping every time this method is called

Reply:
I tried to do this but `Vec<(Gutter, usize)>` doesn't implement the `Debug` trait. This leads to all kinds of problems (e.g. when trying to store it in the `View` struct). Do you know how to solve this?

Reply:
Ah yeah, I think `Debug` is overkill on most of these types, you can derive a custom implementation on `View`:

https://github.com/helix-editor/helix/blob/381dbe70196f28d20c6c0940e62973e029c9765c/helix-view/src/document.rs#L125-L148=

I'd only expose the `id`, `area` and possibly `doc`.

Reply:
Thanks, @archseer , that's good to know! I've implemented the changes.